### PR TITLE
Attempt to reuse 'nameof' expressions in attributes when converting from a local function to a method

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/76188?issue=dotnet%7Croslyn%7C64904

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun @JoeRobich ptal

File: src/Features/CSharpTest/ConvertLocalFunctionToMethod/ConvertLocalFunctionToMethodTests.cs [hunk 1]
```diff
@@ -1115,4 +1115,51 @@ struct S
...
+            {
+                public A(string _) { }
+                public string P { get; set; }
+            }
+
+            class C
+            {
+                void M()
+                {
+                    void [||]M2([A(nameof(b), P = nameof(b))] string b)
```
Comment (ToddGrun) [inline src/Features/CSharpTest/ConvertLocalFunctionToMethod/ConvertLocalFunctionToMethodTests.cs:1136]: >A(nameof(b), P = nameof(b [](http://example.com/codeflow?start=33&length=25)

would it be worth testing that it doesn't use nameof for something for P in the following?

void M2([A(nameof(b), P = "b")] string b)

Comment (CyrusNajmabadi) [reply src/Features/CSharpTest/ConvertLocalFunctionToMethod/ConvertLocalFunctionToMethodTests.cs:1136]: nah :)

and basically i dont' care.  the chance of the user writing that seems miniscule.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**